pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -543,6 +548,54 @@ public SuccessResponse reloadSegmentDeprecated2(
     return reloadSegmentDeprecated1(tableName, segmentName, tableTypeStr);
   }
 
+  @GET
+  @Path("tables/{tableName}/segmentReloadStatus/{taskId}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Get status for a submitted reload operation", notes = "Get status for a submitted reload "
+      + "operation")
+  public ServerReloadTaskStatusResponse getReloadTaskStatus(
```

Comment:
We should consider storing the task id and reload info (table name / segment name) into a ZK node, and add another API to read the reload tasks in the cluster in case user loses the response of the reload call. We can also put other useful info there, e.g. number of messages sent, which can be included in the status check

Reply:
So that API just returns the list of active reload task ids and description like # of messages sent, table names etc?
Agree that might be useful. But should that list be cleaned up by some sort of periodic task? Or no need?

Reply:
Yes. We may put some default retention to keep the reload info (e.g. 1 day). I don't think we need a periodic task to clean up the ZK node. Instead, for each reload call, when we add new entry, we can clean up the expired ones.

Reply:
Although I could not find "retention" specification when creating ZNodes. Is there any example already in place where we might be doing that?

Reply:
I've handled retention via explicitly cleaning up old tasks when new ones are created.